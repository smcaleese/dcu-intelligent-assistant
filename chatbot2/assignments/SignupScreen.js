import React, { useState } from 'react';
import { StyleSheet, View, Text, Alert, TextInput } from 'react-native';
import { Button } from 'react-native-elements';
import axios from 'axios';
import styles from './loginAndSignupStyle';
import CustomButton from '../components/CustomButton';
import { signup } from '../utils/Users';

function CustomInput(props) {
    return (
        <TextInput style={styles.input} placeholder={props.placeholder} value={props.value} 
            onChangeText={text => props.onChangeText(text)} secureTextEntry={props.secureTextEntry} />
    )
}

export default function SignupScreen(props) {

    const [email, setEmail] = useState('')
    const [password, setPassword] = useState('')

    const handleSignup = async () => {
        if(email.length === 0 || password.length === 0) {
            return
        } 
        const signupResponse = await signup(email, password)
        Alert.alert(signupResponse)

        if(signupResponse === "sign up successful") {
            props.setEmail(email)
            goToLogin()    
        }
    }

    const goToLogin = () => {
        props.navigation.navigate('login')
    }

    return (
        <View style={styles.container}>
            <View style={styles.formBox}>
                <View style={styles.titleContainer}>
                    <Text style={styles.titleText}>Signup</Text>
                </View>
                <View style={styles.form}>
                    <CustomInput placeholder="email" value={email} onChangeText={text => setEmail(text)} />
                    <CustomInput placeholder="password" value={password} onChangeText={text => setPassword(text)} secureTextEntry={true} />
                    <View style={{margin: 10}}>
                        <CustomButton title="Submit" color="#4269E1" onPress={() => handleSignup()} />
                        <CustomButton title="Login" color="#bbb" onPress={() => goToLogin()} />
                    </View>
                </View>
            </View>
        </View>
    )
}