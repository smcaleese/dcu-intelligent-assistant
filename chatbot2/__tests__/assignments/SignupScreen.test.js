import React from 'react';
import {create} from 'react-test-renderer';
import SignupScreen from '../../assignments/SignupScreen';

const screen = create(<SignupScreen />)

test('SignupScreen Snapshot Test', () => {
    expect(screen).toMatchSnapshot();
})