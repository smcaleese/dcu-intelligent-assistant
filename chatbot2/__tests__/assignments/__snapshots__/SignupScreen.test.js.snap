// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupScreen Snapshot Test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eee",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 40,
        "width": "85%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#4269E1",
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "padding": 5,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "bold",
            "padding": 5,
            "textAlign": "center",
          }
        }
      >
        Signup
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 5,
          "padding": 5,
          "width": "100%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="email"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "gray",
            "borderRadius": 5,
            "borderWidth": 1,
            "margin": 20,
            "paddingLeft": 20,
            "width": "80%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="password"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "gray",
            "borderRadius": 5,
            "borderWidth": 1,
            "margin": 20,
            "paddingLeft": 20,
            "width": "80%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "margin": 10,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#4269E1",
              "borderRadius": 5,
              "margin": 10,
              "opacity": 1,
              "padding": 15,
              "width": 200,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Submit
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#bbb",
              "borderRadius": 5,
              "margin": 10,
              "opacity": 1,
              "padding": 15,
              "width": 200,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
